Immense.RemoteControl.Desktop.Shared/Services/ScreenCaster.cs

```diff
@@ -180,69 +207,58 @@ private async IAsyncEnumerable<byte[]> GetDesktopStream(IViewer viewer)
                     continue;
                 }
 
-                viewer.PendingSentFrames.Enqueue(new SentFrame(encodedImageBytes.Length, _systemTime.Now));
-
-                var instanceId = Guid.NewGuid();
-                var chunks = encodedImageBytes.Chunk(50_000).ToArray();
-                for (int i = 0; i < chunks.Length; i++)
+                viewer.AppendSentFrame(new SentFrame(encodedImageBytes.Length, _systemTime.Now));
+
+                using var frameStream = _recycleStreams.GetStream();
```

Comment:
Could you figure out the rough general size this stream will be?
`GetStream()` will return a stream with default capacity of (iirc) 4K. If we could initialize this with a size that is close/sightly larger than the frame it will hold, we could reduce the number of re-allocations of this buffer.

Reply:
It will [use the larger](https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/blob/master/src/RecyclableMemoryStream.cs#L245) of the requested size or default block size, which is [128KB](https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/blob/master/src/RecyclableMemoryStreamManager.cs#L58).

I haven't gone through the library's code, but my understand is that the underlying buffers are what get recycled.  When the stream is disposed, the buffer gets released back to the manager's buffer pool.   Since the first frame is usually the largest (since it's full-screen instead of a diff), the buffer(s) would get re-used and wouldn't need to be resized (i.e. add blocks to the pool) for subsequent (smaller) frame diffs.

Anyhow, I tried using `GetStream` with the exact size, and I'm not seeing any difference in memory use or GC frequency.  Do you still recommend adding it anyway?

Reply:
Hmm, that is a good point. My biggest concern was the re-alloc & copying of buffer contents, but if it re-uses the last (largest) buffer then that solves that issue like you mentioned. I think its fine in that case.